packages/components/stamp/src/stamp.tsx

```diff
@@ -21,10 +25,53 @@ type Props = {
    */
   isCondensed: boolean;
   children: ReactNode;
+  icon?: ReactElement;
+  message?: MessageDescriptor & {
+    values?: Record<string, ReactNode>;
+  };
 };
 
 type StylesFunctionParams = Props & { overrideTextColor?: boolean };
 
+const tonesStylesMap = {
```

Comment:
Let's rely on TS to make sure that the keys here are valid.

```suggestion
const tonesStylesMap: Record<Tone, CSSProperties> = {
```

I would also decouple the `iconTone` into a separate function (see other comment).